homework/list.js

```diff
@@ -1,79 +1,83 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
-
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
-
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
-
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
+	var ListManager = function();
+	 this._listNoteContent = [];
+	 this._wrapper = document.querySelector('#note-list-wrapper');
   }
 
-  function updateList(list) {
-    _listNoteContent = list;
-  }
+  ListManager.prototype = {
+  start(){
+  this.fetchList()
+  .then((function)(data){
+  this.updateList(data);
+  this.drawList();
+  this.preloadFirstNote();
+  }).bind(this))
+  .catch((function(){
```

Comment:
Serious issue here: please remember to log **AND** re-throw Promise error. To log it is because you can know the error early and in your own format (also, to prevent it being captured improperly by the following chain), and to re-throw it is to prevent the other chains concating it keep running after the error.